### DISCUSSION TITLE
Is there a document of all information available

### DISCUSSION THREAD

**Discussion by 1512033796** [drewnoakes/metadata-extractor#599] [Q&A]
Is there a document of all information available
for example
![mmexport1673340382854.png](https://user-images.githubusercontent.com/45383053/211503692-bd616794-3cf9-4322-84f4-0c0eff2956fa.png)

**Comment by drewnoakes** [comment]
If you want a document containing all the metadata within a file, you could use this library to create one.

I'm not that clear what you want though.

**Reply by 1512033796** [reply]
I would like to know what information this tool can support to fetch the file, such as the creation time and pixels, and whether there is a dedicated documentation record of the information that can be obtained, preferably with Chinese explanation

**Reply by 1512033796** [reply]
Because I tried to analyze the local image and the server's image, the returned content was different, so I wanted to know what the tool could support

**Reply by drewnoakes** [reply]
There is no document that explains all the tags. There are a lot of tags, and new ones are being added all the time. What's more, not all images have values for all tags, so even if the library supports a given value, there is no guarantee that your image will provide that value.

It would also depend upon what type of file you provide, and what kinds of metadata that file happens to contain.

If you want to see some examples of output from this library, you can look through the `metadata/*.txt` files here:

https://github.com/drewnoakes/metadata-extractor-images/

> preferably with Chinese explanation

This library doesn't have any localisations. I would like to explore adding localisation support in future.

> I tried to analyze the local image and the server's image, the returned content was different

Could you explain more about this?

**Reply by 1512033796** [reply]
> I tried to analyze the local image and the server's image, the returned content was different
Could you explain more about this?

> What's more, not all images have values for all tags, so even if the library supports a given value, there is no guarantee that your image will provide that value.

That's what you mean

In addition, I would like to ask, because I have millions of files here, I want to store all the information that can be obtained by the file in MySQL, but it is not comprehensive to judge what information can be obtained from a file, so I want to know what information this tool supports. How do I save information without all the information files

**Reply by drewnoakes** [reply]
You would run this library against your image then enumerate all the directories, then all the tags. This will show you all the information available in your files that the library supports. If there are additional metadata you require that are not extracted by the library, we can investigate adding support.

**Reply by 1512033796** [reply]
thank you! have a nice day!
